widgets/common/src/DynamicDropdown.jsx

```diff
@@ -1,14 +1,44 @@
 // NOTE: Disabling react/require-default-props as default values are provided in component's definition
 /* eslint-disable react/require-default-props */
+import { useCallback } from 'react';
+import { debounce, isFunction } from 'lodash';
 import VisibilitySensor from 'react-visibility-sensor';
 import './DynamicDropdown.css';
 
 let instanceCount = 0;
+const defaultFetchState = { hasMore: true, currentPage: -1, shouldLoadMore: false };
+
+/**
+ * Creates two `useUpdateEffect` hooks to call `fetchTrigger` function with debouncing.
+ * First hook calls `fetchTrigger` with `reset` argument set to false on `withoutResetFetchDeps` change.
+ * The second calls `fetchTrigger` with `reset` argument set to true on `withResetFetchDeps` change
+ *
+ * @param {function(reset: boolean)} fetchTrigger function to be called to trigger data fetching
```

Comment:
I'm not sure what's this syntax. I've never seen it :smile: Can you tell me what it is?

I've always used TS-compatible syntax in JSDoc

```suggestion
 * @param {(reset: boolean) => void} fetchTrigger function to be called to trigger data fetching
```

Reply:
It's valid JSDoc syntax. I think this can be left as it is.

See my comment: https://github.com/cloudify-cosmo/cloudify-stage/pull/1311#discussion_r614049310 for a bit more explanation.